src/overlays/kaleido_scope/ovl_kaleido_scope/z_kaleido_map.c

```diff
@@ -1,108 +1,963 @@
+    { 255, 255, 0 },
+};
+s16 sWorldMapDotEnvColors[][3] = {
+    { 255, 255, 0 },
+    { 0, 0, 255 },
 };
 
-s16 D_8082B59C[] = {
-    0x0056, 0x0068, 0x0091, 0x0099, 0x00A3, 0x009F, 0x009D, 0x00C7, 0x00D0, 0x00D2, 0x00DA, 0x0000,
+s16 sWorldMapCursorsRectLeft[REGION_MAX] = {
+    86,  // REGION_GREAT_BAY
+    104, // REGION_ZORA_HALL
+    145, // REGION_ROMANI_RANCH
+    153, // REGION_DEKU_PALACE
+    163, // REGION_WOODFALL
+    159, // REGION_CLOCK_TOWN
+    157, // REGION_SNOWHEAD
+    199, // REGION_IKANA_GRAVEYARD
+    208, // REGION_IKANA_CANYON
+    210, // REGION_GORON_VILLAGE
+    218, // REGION_STONE_TOWER
 };
 
-s16 D_8082B5B4[] = {
-    0x007F, 0x0099, 0x008A, 0x00AB, 0x0092, 0x0077, 0x004D, 0x006A, 0x0078, 0x0049, 0x0063, 0x0000,
+s16 sWorldMapCursorsRectTop[REGION_MAX] = {
+    127, // REGION_GREAT_BAY
+    153, // REGION_ZORA_HALL
+    138, // REGION_ROMANI_RANCH
+    171, // REGION_DEKU_PALACE
+    146, // REGION_WOODFALL
+    119, // REGION_CLOCK_TOWN
+    77,  // REGION_SNOWHEAD
+    106, // REGION_IKANA_GRAVEYARD
+    120, // REGION_IKANA_CANYON
+    73,  // REGION_GORON_VILLAGE
+    99,  // REGION_STONE_TOWER
 };
 
-s16 D_8082B5CC[] = {
-    0x0005, 0x0004, 0x0006, 0x0000, 0x0008, 0x0005, 0x0004, 0x0006, 0x0000, 0x0008,
+s16 sGreatFairySpawnRegions[] = {
+    REGION_CLOCK_TOWN, REGION_WOODFALL, REGION_SNOWHEAD, REGION_GREAT_BAY, REGION_IKANA_CANYON,
+    REGION_CLOCK_TOWN, REGION_WOODFALL, REGION_SNOWHEAD, REGION_GREAT_BAY, REGION_IKANA_CANYON,
 };
 
-u16 D_8082B5E0[] = {
-    0xAF, 0xB3, 0xAA, 0xB1, 0xA9, 0xB2, 0xA8, 0xB0, 0xAC, 0xAE,
+void KaleidoScope_DrawWorldMap(PlayState* play) {
+    s16 sceneId;
+    s16 t;
+    s16 n;
+    s16 j;
+    s16 k;
+    s16 i;
+    PauseContext* pauseCtx = &play->pauseCtx;
+    s16 rectLeft;
+    s16 rectRight;
+
+    OPEN_DISPS(play->state.gfxCtx);
+
+    KaleidoScope_SetCursorVtx(pauseCtx, pauseCtx->cursorSlot[PAUSE_MAP] * 4, pauseCtx->mapPageVtx);
+
+    // Draw the world map image
+    if ((pauseCtx->pageIndex == PAUSE_MAP) && (pauseCtx->state == 6) &&
+        ((pauseCtx->unk_200 == 0) || (pauseCtx->unk_200 == 3)) && YREG(6) && (pauseCtx->state != 7) &&
+        !((pauseCtx->state >= 8) && (pauseCtx->state <= 0x12))) {
+
+        // Draw the world map image flat
+        // Because it is flat, the texture is loaded by filling it in 8 rows at a time.
+        // 8 is chosen because it is smaller than `TMEM_SIZE / 2 / WIDTH` and divides the texture's height.
+        // (`TMEM_SIZE / 2` because the texture is color-indexed so the TLUT uses the other half of TMEM.)
+
+        func_8012C628(play->state.gfxCtx);
+
+        gDPSetTextureFilter(POLY_OPA_DISP++, G_TF_POINT);
+        gDPLoadTLUT_pal256(POLY_OPA_DISP++, gWorldMapImageTLUT);
+        gDPSetTextureLUT(POLY_OPA_DISP++, G_TT_RGBA16);
+
+        gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 255, 255, 255, pauseCtx->alpha);
+
+        // Process the 128 rows of pixels for gWorldMapImageTex 8 rows at a time over 16 iterations
```

Comment:
```suggestion
        // Process the 128 rows of pixels for gWorldMapImageTex, 8 rows at a time over 16 iterations
```
Think this is clearer.